ui/src/features/project/pipelines/utils/use-pipelines-infinite-canvas.ts

```diff
@@ -240,16 +236,26 @@ export const usePipelinesInfiniteCanvas = (conf: pipelineInfiniteCanvasHook) =>
     let onWindowMouseMove: (e: MouseEvent) => void = () => {};
 
     const onCanvasMouseDown = (e: MouseEvent) => {
+      // block stage node click
+      let element = e.target as HTMLElement;
+      while (element) {
+        if (element?.classList?.contains(stageNodeClassName)) {
+          return;
+        }
+        element = element.parentNode as HTMLElement;
+      }
```

Comment:
Can't we just handle canvas instead of blocking movement when the stage is detected? It might stop working if we change html structure.

Reply:
Could you expand on it please? I didn't get handle canvas instead blocking movement. stage nodes are in canvas, any mousedown event in stage node will propagate to canvas too and canvas mouse down is interpreted as drag & move.

Reply:
I mean, you can detect when the user clicks the background instead of the whole pipeline. You can add a new div, place it behind the pipeline, and set an absolute position.

Reply:
that is good direction but can we do it in different PR, this hooks need bit refactor anyways as there are lots of moving parts.

Reply:
Can you add a todo comment?